Source/CentralManagerRestoredState.swift

```diff
@@ -60,7 +60,15 @@ public struct CentralManagerRestoredState: CentralManagerRestoredStateType {
         let cbServices = arrayOfAnyObjects.flatMap { $0 as? CBService }
         #endif
 
-        return cbServices.map { Service(peripheral: centralManager.retrievePeripheral(for: $0.peripheral),
-                                        service: $0) }
+        return cbServices.compactMap { cbService -> Service? in
+            do {
+                return Service(
+                    peripheral: centralManager.retrievePeripheral(for: try cbService.unwrapPeripheral()),
+                    service: cbService
+                )
+            } catch {
+                return nil
```

Comment:
should we make a throwable func instead? can we also at least add a log?

Reply:
+1 on throwable

Reply:
Adding a log. The current Swift version does not allow throwing from computed properties and I want to avoid changing the property to a function. Moreover, I think it's better to restore at least some of the services rather than none of the services (if we chose to throw).